### PR TITLE
try and build in two phases to get the build working.

### PR DESCRIPTION
**Pull request by SamBarker**
### Type of change

_Select the type of your PR_

- Bugfix
### Description

Similar to https://github.com/kroxylicious/kroxylicious/pull/240 in an attempt to get lift to be able to build kroxylicious properly.

### Additional Context

_Why are you making this pull request?_

### CODE REVIEW

File: scripts/setupLift.sh [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+set -e
+
+mkdir -p ~/.m2
+
+cat <<EOF > ~/.m2/settings.xml
+<settings xmlns="http://maven.apache.org/SETTINGS/1.0.0"
+  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  xsi:schemaLocation="http://maven.apache.org/SETTINGS/1.0.0
+                      https://maven.apache.org/xsd/settings-1.0.0.xsd">
+  <localRepository>${user.home}/.m2/repository</localRepository>
```
Comment (franvila) [inline scripts/setupLift.sh:18]: Setting `${user.home}` does not work. It provokes this failure
```

Setup's standard error:
/tmp/analyzing-bc07db667a225318/scripts/setupLift.sh: line 13: <settings xmlns="http://maven.apache.org/SETTINGS/1.0.0"
  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
  xsi:schemaLocation="http://maven.apache.org/SETTINGS/1.0.0
                      https://maven.apache.org/xsd/settings-1.0.0.xsd">
  <localRepository>${user.home}/.m2/repository</localRepository>
  <interactiveMode>false</interactiveMode>
  <offline>false</offline>
</settings>
: bad substitution
```
That's why I had to force it to `/home/lift`

Comment (SamBarker) [reply scripts/setupLift.sh:18]: Makes me worry that it's not the right path to use, but let's stick with it for now.

Maybe dump the env in the setup script to see what we can depend on

Comment (franvila) [reply scripts/setupLift.sh:18]: I'm going to try the change suggested by @k-wall in my PR and if it works I will suggest here the same change.

Comment (franvila) [reply scripts/setupLift.sh:18]: It is working fine adding using this approach:

```
M2_HOME=~/.m2

mkdir -p $M2_HOME
cat <<EOF > $M2_HOME/settings.xml
<settings xmlns="http://maven.apache.org/SETTINGS/1.0.0"
  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
  xsi:schemaLocation="http://maven.apache.org/SETTINGS/1.0.0
                      https://maven.apache.org/xsd/settings-1.0.0.xsd">
  <localRepository>$M2_HOME/repository</localRepository>
  <interactiveMode>false</interactiveMode>
  <offline>false</offline>
</settings>
EOF
mvn install -DskipTests -pl :kroxylicious-krpc-plugin -Dquick
```

**Comment by SamBarker** [comment]
Closing in favour of https://github.com/kroxylicious/kroxylicious/pull/244

**Pull request closed without merging**